compiler/rustc_metadata/src/fs.rs

```diff
@@ -24,7 +27,10 @@ pub fn emit_metadata(sess: &Session, metadata: &[u8], tmpdir: &MaybeTempDir) ->
     let result = fs::write(&out_filename, metadata);
 
     if let Err(e) = result {
-        sess.fatal(&format!("failed to write {}: {}", out_filename.display(), e));
+        sess.emit_fatal(FailedWriteError {
+            filename: out_filename.display().to_string(),
+            err: e.to_string(),
```

Comment:
See #100754 for an example of using paths and `io::Error` in the structs directly.

Reply:
✅